h2o-algos/src/main/java/hex/tree/isoforextended/ExtendedIsolationForest.java

```diff
@@ -0,0 +1,377 @@
+
+/**
+ * Extended isolation forest implementation. Algorithm comes from https://arxiv.org/pdf/1811.02141.pdf paper.
+ *
+ * @author Adam Valenta
+ */
+public class ExtendedIsolationForest extends SharedTree<ExtendedIsolationForestModel,
+        ExtendedIsolationForestModel.ExtendedIsolationForestParameters,
+        ExtendedIsolationForestModel.ExtendedIsolationForestOutput> {
+
+    transient IsolationTree[] _iTrees;
+
+    // Called from an http request
+    public ExtendedIsolationForest(ExtendedIsolationForestModel.ExtendedIsolationForestParameters parms) {
+        super(parms);
+        init(false);
+    }
+
+    public ExtendedIsolationForest(ExtendedIsolationForestModel.ExtendedIsolationForestParameters parms, Key<ExtendedIsolationForestModel> key) {
+        super(parms, key);
+        init(false);
+    }
+
+    public ExtendedIsolationForest(ExtendedIsolationForestModel.ExtendedIsolationForestParameters parms, Job job) {
+        super(parms, job);
+        init(false);
+    }
+
+    public ExtendedIsolationForest(boolean startup_once) {
+        super(new ExtendedIsolationForestModel.ExtendedIsolationForestParameters(), startup_once);
+    }
+
+    @Override
+    public void init(boolean expensive) {
+        super.init(expensive);
+        if (_parms.train() != null) {
+            long extensionLevelMax = _parms.train().numCols() - 1;
+            if (_parms.extension_level < 0 || _parms.extension_level > extensionLevelMax) {
+                throw new IllegalStateException("Parameter extension_level must be in interval [0, "
+                        + extensionLevelMax + "] but it is " + _parms.extension_level);
+            }
+            long sampleSizeMax = _parms.train().numRows();
+            if (_parms._sample_size < 0 || _parms._sample_size > sampleSizeMax) {
+                throw new IllegalStateException("Parameter sample_size must be in interval [0, "
+                        + sampleSizeMax + "] but it is " + _parms._sample_size);
+            }
+        }
+    }
+
+    @Override
+    protected double score1(Chunk[] chks, double offset, double weight, double[] fs, int row) {
+        // not used in the current implementation of Extended Isolation Forest
+        throw H2O.unimpl();
+    }
+
+    @Override
+    protected Driver trainModelImpl() {
+        return new ExtendedIsolationForestDriver();
+    }
+
+    @Override
+    public ModelCategory[] can_build() {
+        return new ModelCategory[]{
+                ModelCategory.AnomalyDetection
+        };
+    }
+
+    @Override
+    protected ScoreKeeper.ProblemType getProblemType() {
+        return ScoreKeeper.ProblemType.anomaly_detection;
+    }
+
+    @Override
+    public boolean scoreZeroTrees() {
+        return false;
+    }
+
+    @Override
+    public boolean isSupervised() {
+        return false;
+    }
+
+    @Override
+    public boolean havePojo() {
+        return false;
+    }
+
+    @Override
+    public boolean haveMojo() {
```

Comment:
since MOJO was not really implemented I think its better to return false here and remove the empty MOJOWriter class

Reply:
Resolved in 1a9d25b2d41ce5d77d8a84e221adbba5a5c23e3a